src/websockets/http.py

```diff
@@ -292,6 +292,16 @@ def copy(self) -> "Headers":
         copy._list = self._list.copy()
         return copy
 
+    def get_safe(self, header: str) -> str:
```

Comment:
The only difference between `get_safe` and `__getitem__` is that `get_safe` translates `KeyError` or `MultipleValuesError` into `InvalidHeader`. Having two methods that do the same thing in the same class is weird.

`websockets.http.Headers` is intended as a low-level utility that only knows about HTTP. This is why it uses `websockets.http.MultipleValuesError` instead of one of the websockets-specific exceptions in `websockets.exceptions`.

To factor out this behavior, I'd write a utility function in `websockets.handshake`. The job of this function would be to raise an exception from `websockets.exceptions` and to provide a good error message when the WebSocket protocol specifies that a given header must be provided exactly once, but it's there zero or multiple times.

You can call it `get_unique_header` for example.